op-program/client/l2/engineapi/block_processor.go

```diff
@@ -127,6 +134,10 @@ func (b *BlockProcessor) Assemble() (*types.Block, error) {
 	body := types.Body{
 		Transactions: b.transactions,
 	}
+	if b.dataProvider.Config().IsShanghai(b.header.Number, b.header.Time) {
+		body.Withdrawals = make([]*types.Withdrawal, 0)
+		b.header.WithdrawalsHash = &types.EmptyWithdrawalsHash
+	}
```

Comment:
nit: This is redundant. The engine already does this when the block is assembled. https://github.com/ethereum-optimism/op-geth/blob/f21b97f1d6f673b608f72e9e60d63e70bd925e85/consensus/beacon/consensus.go#L389C1-L399C3

Reply:
That sets the withdrawals in the body, but not the header.  Interestingly it works even if I don't set the header which is making me suspicious again because I can't find where else that is set (in this code path anyway).

Reply:
Ah, it's in `types.NewBlock` in geth.  Excellent, I like not having to reimplement stuff ourself.